src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -86,15 +160,103 @@ public function Accept(\IMapperVisitorRestricted $visitor, $cachingEnabled, \IMa
             $rootNode = new \TdbCmsTree();
             $rootNode->SetLanguage(\TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
             $rootNode->Load($rootTreeId);
+            $visitor->SetMappedValue('breadcrumbStorageHTML', $this->createBreadcrumbStorage($rootNode));
+        }
+
+        $visitor->SetMappedValue('fieldName', $this->fieldName);
+
+        $url = $this->urlUtil->getArrayAsUrl(
+            [
+                'pagedef' => $pagedef,
+                'module_fnc' =>
+                    [
+                        'contentmodule' => 'ExecuteAjaxCall'
+                    ],
+                '_fnc' => 'getTreeNodes',
+                'activeNodeId' => $this->activeNodeId,
+                'rootTreeId' => $rootTreeId,
+                'fieldName' => $this->fieldName,
+                'portalSelectMode' => $portalSelectMode,
+            ],
+            PATH_CMS_CONTROLLER.'?',
+            '&'
+        );
+        $visitor->SetMappedValue('treeNodesAjaxUrl', $url);
+
+        $url = $this->urlUtil->getArrayAsUrl(
+            [
+                'pagedef' => $pagedef,
+                'module_fnc' =>
+                    [
+                        'contentmodule' => 'ExecuteAjaxCall'
+                    ],
+                '_fnc' => 'updateSelection',
+                'table' => $tableName,
+                'currentRecordId' => $currentRecordId,
+                'fieldName' => $this->fieldName
+            ],
+            PATH_CMS_CONTROLLER.'?',
+            '&'
+        );
+        $visitor->SetMappedValue('updateSelectionUrl', $url);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function DefineInterface()
+    {
+        parent::DefineInterface();
+        $externalFunctions =
+            [
+                'getTreeNodes',
+                'updateSelection'
+            ];
+        $this->methodCallAllowed = array_merge($this->methodCallAllowed, $externalFunctions);
```

Comment:
Could be written shorter:

$this->methodCallAllowed[] = 'getTreeNodes';

(+ 1 line)

Reply:
yes true I've changed it.